packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/e2e/fixture.ts

```diff
@@ -30,66 +30,88 @@ type QuanticGeneratedAnswerE2ESearchFixtures = {
-      use
-    ) => {
-      const generatedAnswerObject = new GeneratedAnswerObject(
-        page,
-        data.streamId,
-        analytics,
-        !!options.answerConfigurationId
-      );
-
-      const streamingUrl = options.answerConfigurationId
-        ? ANSWER_API_URL
-        : `${MACHINE_LEARNING_API_URL}/${data.streamId}`;
-      await generatedAnswerObject.mockStreamResponse(
-        streamingUrl,
-        data.streams,
-        data.streamId
-      );
-
-      await page.goto(pageUrl);
-      await configuration.configure(options);
-      await search.waitForSearchResponse();
-
-      generatedAnswerObject.streamEndAnalyticRequestPromise =
-        generatedAnswerObject.waitForStreamEndAnalytics();
-
-      await search.fillSearchInput(exampleQuery);
-      await search.mockSearchWithGenerativeQuestionAnsweringId(data.streamId);
-      await search.performSearch();
-      await search.waitForSearchResponse();
-
-      await use(generatedAnswerObject);
+export const exampleQuery = 'test';
+
+export const testSearch = quanticBase.extend<QuanticGeneratedAnswerE2ESearchFixtures>({
+  pageUrl,
+  genQaData,
+  options: {},
+  analyticsMode: AnalyticsModeEnum.legacy,
+  withFacets: false,
+  afterEach: async ({page}, use) => {
+    await use();
+    await page.unrouteAll({behavior: 'ignoreErrors'});
+  },
+  search: async ({page}, use) => {
+    await use(new SearchObject(page, searchRequestRegex));
+  },
+  generatedAnswer: async (
+    {
+      page,
+      options,
+      configuration,
+      search,
+      genQaData: data,
+      analytics,
+      withFacets,
     },
-  });
+    use
+  ) => {
+    const generatedAnswerObject = new GeneratedAnswerObject(
+      page,
+      data.streamId,
+      analytics,
+      !!options.answerConfigurationId,
+      withFacets
+    );
+
+    const streamingUrl = options.answerConfigurationId
+      ? ANSWER_API_URL
+      : `${MACHINE_LEARNING_API_URL}/${data.streamId}`;
+    await generatedAnswerObject.mockStreamResponse(
+      streamingUrl,
+      data.streams,
+      data.streamId
+    );
+
+    await page.goto(pageUrl);
+    if (withFacets) {
+      await generatedAnswerObject.clickAddFacetsButton();
+    }
+    await configuration.configure(options);
+    await search.waitForSearchResponse();
+
+    generatedAnswerObject.streamEndAnalyticRequestPromise =
+      generatedAnswerObject.waitForStreamEndAnalytics();
+
+    await search.fillSearchInput(exampleQuery);
+    await search.mockSearchWithGenerativeQuestionAnsweringId(data.streamId);
+    await search.performSearch();
+    await search.waitForSearchResponse();
+
+    await use(generatedAnswerObject);
+  },
+});
 
 export const testInsight =
   quanticBase.extend<QuanticGeneratedAnswerE2EInsightFixtures>({
     pageUrl,
     genQaData,
     options: {},
     analyticsMode: AnalyticsModeEnum.legacy,
+    withFacets: false,
+    afterEach: async ({page}, use) => {
```

Comment:
Same here it's very weird to see afterEach in a fixture.

Reply:
yeah I think that we can remove actually